File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/metadata/ConnectionFieldMetadata.java
package com.foundgine.core.semantic.metadata;

public record ConnectionFieldMetadata(String sourceMember, String targetMember, Class<?> sourceType,
		Class<?> targetType, String converter) {
	public ConnectionFieldMetadata(String sourceMember, String targetMember, Class<?> sourceType, Class<?> targetType) {
		this(sourceMember, targetMember, sourceType, targetType, null);
	}
}

File: src/csharp/Foundgine.Core/Semantic/Metadata/ConnectionFieldMetadata.cs
namespace Foundgine.Core.Semantic.Metadata;

/// <summary>
/// Compile-time correspondence between a semantic source member and a target
/// entity member. This is a description only; Foundgine never creates or
/// populates either object.
/// </summary>
public sealed record ConnectionFieldMetadata(
    string SourceMember,
    string TargetMember,
    Type SourceType,
    Type TargetType,
    string? Converter = null);
